clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -1830,6 +1830,58 @@
     public static final Schema[] DELETE_ACLS_REQUEST = {DELETE_ACLS_REQUEST_V0};
     public static final Schema[] DELETE_ACLS_RESPONSE = {DELETE_ACLS_RESPONSE_V0};
 
+    public static final Schema ALTER_REPLICA_DIR_REQUEST_V0 = new Schema(
+        new Field("topics",
+            new ArrayOf(new Schema(
+                new Field("topic", STRING, "The name of the topic."),
+                new Field("partitions", new ArrayOf(new Schema(
+                    new Field("partition", INT32, "The id of the partition."),
+                    new Field("log_dir", STRING, "The absolute log directory path for the partition.")
+                )))
+            ))));
```

Comment:
Sure. The original schema is motivated by the idea that alterReplicaDir operation naturally maps a replica to a log directory. I am not very sure AlterReplicaDirRequest needs to be consistent with DescribeLogDirRequest. But I think what you suggested will make the request more compact and smaller in size. I have made the change as suggested. Thanks!